MIGRATION_GUIDE.md

```diff
@@ -1,9 +1,170 @@
 across different versions.
 
+## vX.XX.X -> v0.85.0
+
+### Migration from old grant resources to new ones
+
+In recent changes, we introduced new grant resources intended to replace old grant solutions. Here's some of the useful
+information that may help during the migration of the grant resources. Here's an example of the `snowflake_database_grant` to `snowflake_grant_privileges_to_account_role` migration.
+The migration can be done in two ways. Either you can remove old grant resources and replace them with new ones or perform
+more complicated migration, but without revoking any grant (no downtime migration). We'll focus on the second one as the first approach
+is pretty straight forward. As an example we'll take `snowflake_database_grant` that grants one privilege to two roles:
+
+```terraform
+resource "snowflake_database_grant" "old_resource" {
+  depends_on = [ snowflake_database.test, snowflake_role.a, snowflake_role.b ]
+  database_name = snowflake_database.test.name
+  privilege = "USAGE"
+  roles = [ snowflake_role.a.name, snowflake_role.b.name ]
+}
+```
+
+#### 1. terraform list
+
+Run `terraform state list` to search the grants you're looking for (for larger configurations it's best to filter the results), 
+for example, `terraform state list | grep "snowflake_database_grant"`.
+
+#### 2. terraform rm
+
+Now choose which one you would to migrate next and remove it from the state with `terraform state rm <resource_address>`. 
+In our example, `terraform state rm snowflake_database_grant.old_resource`. After running the command, you can remove the resource from the configuration 
+(removing the state will "detach" it from the resource block, so after removing it, the Terraform won't try to revoke USAGE from our roles).
+
+#### 3. Two options from here
+
+At this point, we have several options for creating new grant resources that will replace the old ones.
+We will cover three options:
+- Configuration + Terraform CLI
+- Configuration + import block
+- Generating the configuration with import block and `terraform plan -generate-config-out`
+
+#### 3.1.1. Write a new grant resource that will be an equivalent of the older one
+
+```terraform
+resource "snowflake_grant_privileges_to_account_role" "new_resource" {
+  depends_on = [snowflake_database.test, snowflake_role.a, snowflake_role.b]
+  for_each   = toset([snowflake_role.a.name, snowflake_role.b.name])
+  privileges = ["USAGE"]
+  role_name  = each.key
+  on_account_object {
+    object_type = "DATABASE"
+    object_name = snowflake_database.test.name
+  }
+}
+```
+
+#### 3.1.2. terraform import
+
+Write the `terraform import` command with the ID so that the resource will be able to parse and fill the state correctly.
+You can find import syntax in the documentation for a given resource, [here](https://registry.terraform.io/providers/Snowflake-Labs/snowflake/latest/docs/resources/grant_privileges_to_account_role#import)
```

Comment:
would also put a link to id examples: https://registry.terraform.io/providers/Snowflake-Labs/snowflake/latest/docs/resources/grant_privileges_to_account_role#import-examples